core/src/main/java/org/pentaho/platform/scheduler2/action/SchedulerOutputPathResolver.java

```diff
@@ -119,7 +155,7 @@ private String runAsUser( Callable<String> callable ) {
     return null;
   }
 
-  private boolean isValidOutputPath( String path ) throws SchedulerException {
+  protected boolean isValidOutputPath( String path ) throws SchedulerException {
```

Comment:
While this is fine as is, in other cases, we tried using the `IGenericFileService` directly in the CE version, making the code more uniform.

Reply:
There is other specific JCR repository logic. More specifically checking the meta data of a repository file. Does `IGenericFileService` support reading and writing metadata

For full context:


```
  protected boolean isValidOutputPath( String path ) throws SchedulerException {
    try {
      RepositoryFile repoFile = getRepository().getFile( path );
      if ( repoFile != null && repoFile.isFolder() ) {
        boolean scheduleAllowed = isScheduleAllowed( repoFile.getId() );
        if ( scheduleAllowed ) {
          return true;
        } else {
          throw new SchedulerException( Messages.getInstance().getString(
            "QuartzScheduler.ERROR_0009_SCHEDULING_IS_NOT_ALLOWED_AFTER_CHANGE", getJobName(), this.actionUser ) );
        }
      }
    } catch ( Exception e ) {
      logger.warn( e.getMessage(), e );
    }
    return false;
  }

... ... ...

  private boolean isScheduleAllowed( final Serializable repositoryId ) {
    boolean canSchedule = false;
    canSchedule = getAuthorizationPolicy().isAllowed( SCHEDULER_ACTION_NAME );
    if ( canSchedule ) {
      Map<String, Serializable> metadata = getRepository().getFileMetadata( repositoryId );
      if ( metadata.containsKey( RepositoryFile.SCHEDULABLE_KEY ) ) {
        canSchedule = BooleanUtils.toBoolean( (String) metadata.get( RepositoryFile.SCHEDULABLE_KEY ) );
      }
    }

    return canSchedule;
  }
```


what would you suggest, if any changes?

Reply:
Not currently, no. We'd have to generalize the concepts in some way. Related comment on the EE pr.
Not something for now.